### ISSUE TITLE
[rebasing] Premature force push warning?

### ISSUE THREAD

**Issue by tierninho** [desktop/desktop#7341]
During a rebase we show a force push warning for the user in anticipation of the Force Push button appearing. Under the hood are we still force pushing if the rebase was successful? 

I still see this warning even though the success messaging below is shown. The Force Push button is never shown in this case, so I am not sure if this is intended or not.

<img width="611" alt="Screen Shot 2019-04-17 at 3 37 52 PM" src="https://user-images.githubusercontent.com/14828183/56331236-3017dd80-6127-11e9-98ef-0af386a397da.png">
<img width="817" alt="Screen Shot 2019-04-17 at 3 37 59 PM" src="https://user-images.githubusercontent.com/14828183/56331235-3017dd80-6127-11e9-8ee1-9b6a0dfe343a.png">

Secondly, when the user actually does click the Force Push button (if pref enabled), we show another warning like so. Curious, do we really need both warnings? 

<img width="645" alt="Screen Shot 2019-04-17 at 3 49 23 PM" src="https://user-images.githubusercontent.com/14828183/56331569-918c7c00-6128-11e9-8790-22a37a377950.png">

1.7.0-beta4, mac

**Comment by billygriffin** [comment]
@tierninho I'm not sure how to answer the first question (cc: @shiftkey), but for the second one, that's intentional for now because they're different kinds of problems (one is getting into a potentially tricky spot for yourself locally, and the other is potentially messing up the bigger repo for everyone on the team). It's a matter of who it impacts.

**Comment by shiftkey** [comment]
I think you're interested in this TODO comment, which I deferred from the initial beta:

https://github.com/desktop/desktop/blob/ab7b0e9bfedd27834fa2fd3b8f49f11c0f8dcafc/app/src/lib/stores/app-store.ts#L3922-L3924

Consider this scenario:

 - you rebase a branch cleanly onto the base branch
 - you try to do the exact same thing again

There's no work to do here for a rebase, but we don't perform this check. I think implementing that check and blocking the user from proceeding is the answer here.

Can you confirm that was the steps you were testing?

**Comment by shiftkey** [comment]
> Curious, do we really need both warnings?

I'm inclined to keep them both around for the moment, because they feel like distinct cases:

 - an up-front warning before you dive into the rebase of the downstream impact to others
 - a warning before pushing to give you a chance to back out

**Comment by tierninho** [comment]
> Consider this scenario:
you rebase a branch cleanly onto the base branch
you try to do the exact same thing again

> There's no work to do here for a rebase, but we don't perform this check. I think implementing that check and blocking the user from proceeding is the answer here.

> Can you confirm that was the steps you were testing?

The check sounds good. Yes, this is likely the flow I took. 👍

**Comment by shiftkey** [comment]
@billygriffin if you're still fine with the dual warnings before starting the rebase and before force-pushing I think we can close this out for now...

**Comment by billygriffin** [comment]
Yup, I'm all set, thanks!
